### ISSUE TITLE
enterprise.splunk.com/v1alpha1 resources spec is too generic

### ISSUE THREAD

**Issue by nickperry** [splunk/splunk-operator#14]
```apiVersion: enterprise.splunk.com/v1alpha1
kind: SplunkEnterprise
metadata:
  name: example
spec:
  resources:
    ...
```

This spec is mostly generic and not per instance type. I need different sizing for indexers, search-heads, spark workers and ICMs.

For example, I would like

```splunkCpuRequest```

to be split up into ```
indexerCpuRequest
searchHeadCpuRequest```
etc.

**Comment by mikedickey** [comment]
Planning to address this as part of #17

**Comment by mikedickey** [comment]
Merged to develop for inclusion in 0.1.0. See new v1alpha2 CRDs.
